include/deal.II/numerics/smoothness_estimator.h

```diff
@@ -0,0 +1,385 @@
+   * decide whether to perform h- or p-adaptation. So in most cases, we only
+   * need to calculate those indicators on cells flagged for refinement or
+   * coarsening. This parameter controls whether this particular subset or all
+   * cells will be considered. By default, only flagged cells will be
+   * considered: Smoothness indicators will only be set on those vector entries
+   * of flagged cells; the others will be set to zero.
+   *
+   * For more theoretical details see
+   * @code{.bib}
+   * @Article{Mavriplis1994,
+   *  author    = {Mavriplis, Catherine},
+   *  title     = {Adaptive mesh strategies for the spectral element method},
+   *  journal   = {{Computer Methods in Applied Mechanics and Engineering}},
+   *  year      = {1994},
+   *  volume    = {116},
+   *  number    = {1},
+   *  pages     = {77--86},
+   *  publisher = {Elsevier},
+   * }
+   * @article{Houston2005,
+   *  author    = {Houston, Paul and S{\"u}li, Endre},
+   *  title     = {A note on the design of hp-adaptive finite element
+   *               methods for elliptic partial differential equations},
+   *  journal   = {{Computer Methods in Applied Mechanics and Engineering}},
+   *  number    = {2},
+   *  pages     = {229--243},
+   *  publisher = {Elsevier},
+   *  volume    = {194},
+   *  year      = {2005}
+   * }
+   * @article{Eibner2007,
+   *  author    = {Eibner, Tino and Melenk, Jens Markus},
+   *  title     = {An adaptive strategy for hp-FEM based on testing for
+   *               analyticity},
+   *  journal   = {{Computational Mechanics}},
+   *  year      = {2007},
+   *  volume    = {39},
+   *  number    = {5},
+   *  pages     = {575--595},
+   *  publisher = {Springer},
+   * }
+   * @endcode
+   * and for the application within the deal.II:
+   * @code{.bib}
+   * @article{Davydov2017,
+   *   author  = {Denis Davydov and Tymofiy Gerasimov and Jean-Paul Pelteret and
+   *              Paul Steinmann},
+   *   title   = {Convergence study of the h-adaptive PUM and the hp-adaptive
+   *              FEM applied to eigenvalue problems in quantum mechanics},
+   *   journal = {{Advanced Modeling and Simulation in Engineering Sciences}},
+   *   year    = {2017},
+   *   volume  = {4},
+   *   number  = {1},
+   *   pages   = {7},
+   *   issn    = {2213-7467},
+   *   doi     = {10.1186/s40323-017-0093-0},
+   * }
+   * @endcode
+   *
+   * @ingroup numerics
+   * @author Denis Davydov, 2018
+   */
+  template <int dim, int spacedim, typename VectorType>
+  void
+  legendre_coefficient_decay(
+    FESeries::Legendre<dim, spacedim> &    fe_series,
+    const hp::DoFHandler<dim, spacedim> &  dof_handler,
+    const std::vector<const VectorType *> &all_solutions,
+    std::vector<Vector<float> *> &         all_smoothness_indicators,
+    const std::function<void(std::vector<bool> &flags)>
+      &coefficients_predicate = [](std::vector<bool> &flags) -> void {
+      std::fill(flags.begin(), flags.end(), true);
```

Comment:
This seems complicated. Why not just take a `vector<bool>` that either has the right size or has size zero where the latter is interpreted as a vector of all `true`? You could even abuse the `ComponentMask` class for this purpose.

Reply:
This code also seems to have moved. What did you decide?